packages/labs/analyzer/src/test/lit-element/properties_test.ts

```diff
@@ -211,5 +212,53 @@ for (const lang of languages) {
     assert.equal(property.attribute, 'static-prop');
   });
 
+  test('property with an unsupported name type', ({analyzer, element}) => {
+    const diagnostics = Array.from(analyzer.getDiagnostics());
+    assert.equal(diagnostics.length, 1);
+    assert.equal(
+      diagnostics[0].code,
+      DiagnosticCode.UNSUPPORTED_PROPERTY_NAME_TYPE
+    );
+
+    // Fields named with symbols are visible in the `fields` iterator.
+    const field = Array.from(element.fields).find(
+      (x) => x.name === '[unsupportedPropertyName]'
+    );
+    assert.ok(field);
```

Comment:
I'm not sure if this is a bug or not. Symbol-named properties make it into `fields` here:

https://github.com/lit/lit/blob/be18292938062a3b5233016fdac1a72ba6f1eacf/packages/labs/analyzer/src/lib/javascript/classes.ts#L89-L100


...which doesn't seem to care what type of name a property is using, but I'm not sure if that was intentional or not. `node.name` is the AST node for however the name of a field is described, so its `getText()` ends up returning the name with brackets around it, which looks reasonable but is really just JS text. (I think `getText()` would return `["abc" + "def"]` for a field like `class { ["abc" + "def"] = 123; }`.)

Reply:
hmm...is that a bug? What do we want the name to be? I could see that for the text representation, that something like `[mySymbol]` is about as good as we can get. But to fully support Symbol-named fields we probably want to represent them as a reference in the manifest, so we'd need to keep the name AST node.

I think that symbol-named fields are rare enough that we could punt on a few things, keep issues, then come back with a push to support symbols better, including adding name reference support to ClassField and ClassMethod: https://github.com/webcomponents/custom-elements-manifest/blob/main/schema.d.ts#L557

Reply:
I've updated this to also log a diagnostic rather than analyzing the property.